docs/dev/publicdashboard/Working_with_public_dashboard.md

```diff
@@ -0,0 +1,159 @@
+1. Front end testing
+    - Make sure the dropdown menu have valid list of metrics.
+    - Make sure the default charts are loaded at the first instance of loading the website.
+    - Make sure all the charts are being launched or the error table is displayed.
+2. Test with loading and unloading different program/study dataset. 
+    - Load different datasets, run the scripts to execute the notebooks to make sure the charts are being generate properly.
+    - Between datasets, drop the old dataset and load the new dataset.
+    - Also check with empty MongoDB.
+3. After the code has been merged, validate the changes on the Staging.
+    - Generate a snapshot of the charts from the staging website.
+    - Request for the dataset for staging-program.
+    - Validate the changes for both the charts and the Front end aspect.
+
+# Different docker-compose yml files?
+We have two docker-compose yml files, one specifically to aid in running in development mode. Another to run in production - alike mode.
+
+1. About the `docker-compose.dev.yml` file:
+ - This is useful in case, you want to launch the Jupyter notebook and execute the code block and explore output of each code blocks.
+ ```
+services:
+    notebook-server:
+        environment:
+        - DB_HOST=db
+        - WEB_SERVER_HOST=0.0.0.0
+        - CRON_MODE=
+        - STUDY_CONFIG=stage-program
+```
+
+To launch docker-compose.dev.yml:
+```
+docker-compose -f docker-compose.dev.yml up
+```
+
+2. About the `docker-compose.yml` file:
+ - This is useful in case, you want to execute the scripts to run all the required notebooks, without making any changes in the existing notebooks.
+ - Note: We use cron jobs to execute the different notebooks in a scheduled way daily to generate the charts.
+ ```
+services:
+    notebook-server:
+        environment:
+        - DB_HOST=db
+        - WEB_SERVER_HOST=0.0.0.0
+        - CRON_MODE=TRUE
+        - STUDY_CONFIG=stage-program
+```
+
+To launch docker-compose.yml:
+```
+docker-compose -f docker-compose.yml up
+```
+
+ Key difference amongst these two docker-compose file is the `CRON_MODE=TRUE` in case of `docker-compose.yml` while its disabled for `docker-compose.dev.yml`.
+
+# Changes required in docker-compose.dev.yml and/or docker-compose.yml to load different dataset
+
+The default `docker-compose.yml` looks like:
+```
+services:
+    notebook-server:
+        environment:
+        - DB_HOST=db
+        - WEB_SERVER_HOST=0.0.0.0
+        - CRON_MODE=TRUE
+        - STUDY_CONFIG=stage-program
+```
+This will load the dataset saved in the MongoDB as `Stage_database`.
+
+What if we want to load a different dataset than staging, and test for a different program/study.
+Let us consider the example below:
+```
+services:
+    notebook-server:
+        environment:
+        - DB_HOST=mongodb://db/openpath_prod_usaid_laos_ev
+        - WEB_SERVER_HOST=0.0.0.0
+        - CRON_MODE=TRUE
+        - STUDY_CONFIG=usaid-laos-ev
```

Comment:
This may become obsolete once the changes discussed in https://github.com/e-mission/e-mission-docs/issues/1048 are checked in.

Reply:
Since, there's still progress ongoing for changes related to #1048 , I have included a note to update this once the changes re checked-in.

Reply:
#1048 is being actively worked on, and will be merged in the next month or so. And the next time it is updated, this will need to be updated as well. 

> Bad/obsolete documentation is just as bad as no documentation.

Why does this need to be replicated here?

Reply:
>Why does this need to be replicated here?

At present, while documenting the process of working with Public Dashboard, this is a rightful approach of changing the dataset and study config in the docker-compose file. Therefore, I felt it would be rightful to keep this part of documentation (provided reference that change would be need to be made when the #1048 is done). Considering document would be updated when changes are made on the the referred issue.

But accounting for the future prospect of being obsolete, the above block of instructions have been removed.

Reply:
I think that it is fine to showcase how to change the values. But you can omit the parts that have not changed - e.g. 

```
        environment:
        - DB_HOST=mongodb://db/openpath_prod_usaid_laos_ev
        - ....
        - STUDY_CONFIG=usaid-laos-ev
```

Reply:
Thank you for the clarification, I have updated  the section showcasing the example of how to change the values for DB_HOST and STUDY_CONFIG, while omitting the parts which have not been changed.